### PR TITLE
Bug 1205800 - Added animations to transitioning to PBMode

### PR DESCRIPTION
**Pull request by sleroux**

### CODE REVIEW

**Comment by thebnich** [comment]
These are pretty sweet animations. Sometimes when switching from private -> normal mode, though, I see a flicker that causes the thumbnails to jump. Do you see this, too?

**Comment by sleroux** [comment]
@thebnich Are you seeing the flicker on the tabs animation or the toggle button? I've updated the toggle button to use a different method of animation and made it circle so that might have fixed any flickering with that. For the tabs, I couldn't replicate any flickering. Which device are you seeing this on?

Also, I think @darrinhenein is revising the tabs animation so it might change.

**Comment by thebnich** [comment]
Was seeing flicker on the tab thumbnail, but that seems to be gone now with this PR :+1: Was just using the simulator.

File: Client/Frontend/Browser/TabTrayController.swift [hunk 1]
```diff
@@ -235,9 +235,8 @@ class TabTrayController: UIViewController {
     private var privateMode: Bool = false {
         didSet {
             if #available(iOS 9, *) {
-                togglePrivateMode.selected = privateMode
+                togglePrivateMode.selected = !togglePrivateMode.selected
```
Comment (thebnich) [inline Client/Frontend/Browser/TabTrayController.swift]: This change seems weird...why wouldn't we want `togglePrivateMode.selected` to always match `privateMode`?

File: Client/Frontend/Widgets/ToggleButton.swift [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    private func updateMaskPathForSelectedState(selected: Bool) {
+        let path = CGPathCreateMutable()
+        if selected {
+            CGPathAddEllipseInRect(path, nil, CGRect(origin: CGPointZero, size: maskShapeLayer.bounds.size))
+            self.maskShapeLayer.path = path
+        } else {
+            CGPathAddEllipseInRect(path, nil, CGRect(origin: maskShapeLayer.position, size: CGSizeZero))
+            self.maskShapeLayer.path = path
```
Comment (thebnich) [inline Client/Frontend/Widgets/ToggleButton.swift]: Could move this outside of the `if` if you wanted.

File: Client/Frontend/Browser/TabTrayController.swift (hunk 1 above)
Comment (sleroux) [inline Client/Frontend/Browser/TabTrayController.swift]: Ya not sure what happened here. Reverting this.

**Pull request merged**